kerastuner/tuners/hyperband.py

```diff
@@ -102,25 +113,21 @@ def _run_values(self, space, trial_id):
         self._trial_id_to_candidate_index[trial_id] = candidate_index
         if candidate is not None:
             values = self._copy_values(space, candidate)
-            values['tuner/trial_id'] = self._index_to_id[candidate_index]
+            if trial_id != self._index_to_id[candidate_index]:
+                values['tuner/trial_id'] = self._index_to_id[candidate_index]
             return {'status': 'RUN', 'values': values}
         return {'status': 'EXIT'}
 
     @staticmethod
     def _copy_values(space, values):
-        return_values = {}
+        return_values = values.copy()
         for hyperparameter in space:
-            if hyperparameter.name in values:
-                return_values[
-                    hyperparameter.name] = values[hyperparameter.name]
-            else:
+            if hyperparameter.name not in values:
                 return_values[hyperparameter.name] = hyperparameter.default
```

Comment:
It's `default_value`, not `default` -- please add test coverage for this code path.